hydradx/other tests/test_price_manipulations.py

```diff
@@ -0,0 +1,479 @@
+
+        agent_prices = {
+            ('omnipool', tkn_buy): initial_price
+        }
+
+        initial_agent = Agent(
+            holdings=agent_holdings,
+            share_prices=agent_prices,
+        )
+
+        # trade to manipulate the price
+        first_sell = initial_state.liquidity[tkn_buy] * min(
+            initial_state.trade_limit_per_block,
+            initial_state.max_withdrawal_per_block,
+            initial_state.remove_liquidity_volatility_threshold
+        )
+
+        new_state, new_agent = oamm.execute_swap(
+            state=initial_state.copy(),
+            agent=initial_agent.copy(),
+            tkn_sell=tkn_sell,
+            tkn_buy=tkn_buy,
+            sell_quantity=first_sell
+        )
+        first_buy = new_agent.holdings[tkn_buy] - new_agent.holdings[tkn_buy]
+
+        oamm.execute_remove_liquidity(
+            state=new_state,
+            agent=new_agent,
+            quantity=new_agent.holdings[('omnipool', tkn_buy)],
+            tkn_remove=tkn_buy
+        )
+
+        tkn_buy, tkn_sell = tkn_sell, tkn_buy
+        market_prices = {tkn: oamm.usd_price(initial_state, tkn) for tkn in initial_state.asset_list}
+        second_sell = (math.sqrt((
+            new_state.lrna[tkn_sell] * new_state.lrna[tkn_buy]
+            * new_state.liquidity[tkn_sell] * new_state.liquidity[tkn_buy]
+        ) / (market_prices[tkn_buy] / market_prices[tkn_sell])) - (
+            new_state.lrna[tkn_buy] * new_state.liquidity[tkn_sell]
+        )) / (new_state.lrna[tkn_sell] + new_state.lrna[tkn_buy])
+
+        final_state, final_agent = oamm.execute_swap(
+            state=new_state,
+            agent=new_agent,
+            tkn_sell=tkn_sell,
+            tkn_buy=tkn_buy,
+            sell_quantity=second_sell
+        )
+        second_buy = final_agent.holdings[tkn_buy] - new_agent.holdings[tkn_buy]
+
+        profit = (
+                oamm.cash_out_omnipool(final_state, final_agent, market_prices)
+                - oamm.cash_out_omnipool(initial_state, initial_agent, market_prices)
+        )
+
+        if profit > max_profit[0]:
+            max_profit = (profit, lp_quantity, initial_price, first_buy, second_buy, tkn_buy, tkn_sell)
+
+    if max_profit[0] > 5:
+        attack_asset = max_profit[6]
+        trade_asset = max_profit[5]
+        print(f'max profit:')
+        print(f'agent LPs {max_profit[1]} {attack_asset} ({max_profit[1] / initial_state.liquidity[attack_asset] * 100}'
+              f'%) at {max_profit[2]} USD spot price.')
+        print(f'agent buys {max_profit[3]} {attack_asset} with {trade_asset}')
+        print(f'agent withdraws all {attack_asset}')
+        print(f'agent buys {max_profit[4]} {trade_asset} with {attack_asset}')
+        print(f'agent nets {max_profit[0]}')
+        print(f'initial omnipool state: {initial_state}')
+
+
+# @settings(max_examples=10000)
+def test_withdraw_manipulation_scenario():
```

Comment:
Is it valuable to have this specific scenario in our tests? Where is it from?

Reply:
I think we can actually get rid of this test. The reason it exists is because I wanted to double-check scenarios that were coming out as profitable in the fuzz tests, but that is very specific to the situation at the time and doesn't seem relevant now.